src/probnum/filtsmooth/statespace/transition.py

```diff
@@ -36,29 +36,18 @@ class Transition(abc.ABC):
         Discretely indexed transitions (transformations)
     """
 
-    def __call__(
-        self,
-        arr_or_rv: Union[np.ndarray, "RandomVariable"],
-        start: float = None,
-        stop: float = None,
-        **kwargs
-    ) -> ("RandomVariable", Dict):
-        """
-        Transition a random variable or a realization of one.
-
-        The input is either interpreted as a random variable or as a realization.
-        Accordingly, the respective methods are called: :meth:`transition_realization` or :meth:`transition_rv`.
-        """
-        if isinstance(arr_or_rv, RandomVariable):
-            return self.transition_rv(rv=arr_or_rv, start=start, stop=stop, **kwargs)
-        return self.transition_realization(
-            real=arr_or_rv, start=start, stop=stop, **kwargs
-        )
+    def __init__(self):
+        self.precon = None
 
     @abc.abstractmethod
     def transition_realization(
-        self, real: np.ndarray, start: float, stop: float = None, **kwargs
-    ) -> ("RandomVariable", Dict):
+        self,
+        real: np.ndarray,
+        start: float,
+        stop: Optional[float] = None,
+        step: Optional[float] = None,
+        linearise_at: Optional[RandomVariable] = None,
+    ) -> (RandomVariable, Dict):
```

Comment:
all possible positional arguments are tracked in this interface. I hope this makes it more clear, which kwargs are possible and which arent.